From 6ed73f150c99677ad3822f968eb8e553a8d1c4f1 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@gmail.com>
Date: Fri, 17 May 2013 14:11:44 +0200
Subject: [PATCH] spice-session: use a more robust signal connect

The Spice session may outlive the virt-viewer session, due to it's async
nature. Use the more robust virt_viewer_signal_connect_object() to fix
delayed potential crashes.
---
 src/virt-viewer-session-spice.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/virt-viewer-session-spice.c b/src/virt-viewer-session-spice.c
index d7b01a7..06e0005 100644
--- a/src/virt-viewer-session-spice.c
+++ b/src/virt-viewer-session-spice.c
@@ -194,10 +194,10 @@ create_spice_session(VirtViewerSessionSpice *self)
     self->priv->gtk_session = spice_gtk_session_get(self->priv->session);
     g_object_set(self->priv->gtk_session, "auto-clipboard", TRUE, NULL);
 
-    g_signal_connect(self->priv->session, "channel-new",
-                     G_CALLBACK(virt_viewer_session_spice_channel_new), self);
-    g_signal_connect(self->priv->session, "channel-destroy",
-                     G_CALLBACK(virt_viewer_session_spice_channel_destroy), self);
+    virt_viewer_signal_connect_object(self->priv->session, "channel-new",
+        G_CALLBACK(virt_viewer_session_spice_channel_new), self, 0);
+    virt_viewer_signal_connect_object(self->priv->session, "channel-destroy",
+        G_CALLBACK(virt_viewer_session_spice_channel_destroy), self, 0);
 
     manager = spice_usb_device_manager_get(self->priv->session, NULL);
     if (manager) {
